repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -3298,20 +3281,51 @@ public List<String> deleteClassificationPropagation(String entityGuid, String cl
             }
 
             AtlasClassification classification = entityRetriever.toAtlasClassification(classificationVertex);
-            List<AtlasVertex> entityVertices = deleteDelegate.getHandler().removeTagPropagation(classificationVertex);
-            deleteDelegate.getHandler().deleteClassificationVertex(classificationVertex, true);
-            if (CollectionUtils.isEmpty(entityVertices)) {
+
+            List<AtlasEdge> propagatedEdges = getPropagatedEdges(classificationVertex);
+            if (propagatedEdges.isEmpty()) {
+                LOG.warn("deleteClassificationPropagation(classificationVertexId={}): classification edges empty", classificationVertexId);
+
                 return null;
             }
 
-            List<String> impactedGuids = entityVertices.stream().map(x -> GraphHelper.getGuid(x)).collect(Collectors.toList());
-            GraphTransactionInterceptor.lockObjectAndReleasePostCommit(impactedGuids);
+            List<String> deletedPropagationsGuid = new ArrayList<>();
+            long propagatedEdgesSize = propagatedEdges.size();
+            int toIndex;
+            int offset = 0;
+
+            LOG.info(String.format("Number of edges to be deleted : %s for classification vertex with id : %s", propagatedEdgesSize, classificationVertexId));
 
-            List<AtlasEntity>   propagatedEntities = updateClassificationText(classification, entityVertices);
+            do {
+                toIndex = ((offset + CHUNK_SIZE > propagatedEdgesSize) ? (int) propagatedEdgesSize : (offset + CHUNK_SIZE));
 
-            entityChangeNotifier.onClassificationsDeletedFromEntities(propagatedEntities, Collections.singletonList(classification));
+                List<AtlasVertex> entityVertices = deleteDelegate.getHandler().removeTagPropagation(classification, propagatedEdges.subList(offset, toIndex));
+
+                if (CollectionUtils.isEmpty(entityVertices)) {
+                    return null;
```

Comment:
don't we need to continue for rest of chunks although intermediate chunks have entity vertices that have already been removed?

Reply:
Yes we need to continue here. Have to fix it

Reply:
Line:3304 if (CollectionUtils.isEmpty(entityVertices)) {...

I think no need for this check given that propagatedEdges will never be empty at this point & so deleteDelegate.getHandler().removeTagPropagation(..) must return non-empty list

Reply:
@sumandas0  are you fixing it or should I update?

Reply:
Yes I've fixed that